internal/pkg/service/buffer/api/http/http.go

```diff
@@ -80,19 +80,20 @@ func HandleHTTPServer(proc *servicectx.Process, d dependencies.ForServer, u *url
 
 	proc.Add(func(ctx context.Context, shutdown servicectx.ShutdownFn) {
 		// Start HTTP server in a separate goroutine.
-		go func() {
-			logger.Infof("HTTP server listening on %q", u.Host)
-			shutdown(srv.ListenAndServe())
-		}()
+		logger.Infof("HTTP server listening on %q", u.Host)
+		shutdown(srv.ListenAndServe())
+	})
 
-		// Wait for termination
-		<-ctx.Done()
+	proc.OnShutdown(func() {
 		logger.Infof("shutting down HTTP server at %q", u.Host)
```

Comment:
- pri vypinanie API node by sme mali syncnut statistiky
- musime to ale spravit az po vypnuti HTTP servera, pocas graceful shutdown

`OnShutdown` logika je LIFO:
- HTTP server sa spusta ako posledny, cim odblokuje `/health` check v K8S a zaradi sa do load-balancer
- HTTP server potrebujeme vypinat ako prvy, ... kedze az ked dobehne posledny request, tak mozeme vypinat ostatne sluzby, napr. `statistics`.

Shutdown HTTP servera nebol vyrieseny cez `OnShutdown` (`LIFO`) a neblokoval ostatne shutdown calls, takze som to fixol.

Reply:
Da sa to skontrolovat aj tym, ze si API node spustim manualne:
![image](https://user-images.githubusercontent.com/19371734/208068526-22cc89df-d440-4fa5-aea8-70972de3cb6c.png)